cargo-cyclonedx/src/generator.rs

```diff
@@ -757,7 +766,8 @@ impl GeneratedSbom {
                     Pattern::Bom | Pattern::Package => unreachable!(),
                 }
             })
-            .map(|(component, target_kind)| {
+            .map(|component| {
+                let target_kind = &target_kinds.0[component.bom_ref.as_ref().unwrap()];
```

Comment:
It's possible to only perform one hash table lookup instead of one, but it's not worth the trouble - there are only a handful of these calls, and it's not worth the additional complexity this would introduce with a `filter_map` call.

Reply:
Agreed